compiler-core/src/ast.rs

```diff
@@ -1643,6 +1666,10 @@ pub enum Pattern<Type> {
         location: SrcSpan,
         name: EcoString,
         type_: Type,
+        /// We need to keep track of this at the variable pattern level in order
```

Comment:
What's a variable pattern level?

Reply:
Oops poor wording on my part, what I meant to say is we need* to store this information directly on the `Pattern::Variable` node instead on the `CallArg` container to give better error messages

\* Actually there could be a way to do this without duplicating this information on the pattern node as well but it would be a pretty big signature change to the `unify` function and as I was trying that, it made the code much harder to read. So I opted for the smallest change here.
But if you prefer I can show you the other approach as well!

Reply:
It's not data about the variable so it should not be on the variable node.